src/Statistics/Acquisition/StatisticsRequest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace AppBundle\Statistics\Acquisition;
+
+use AppBundle\Utils\DateTimeFactory;
+use Symfony\Component\HttpFoundation\Request;
+
+class StatisticsRequest
+{
+    private const DATE_FORMAT = 'd-m-Y';
+
+    private $tags;
+    private $startDate;
+    private $endDate;
+
+    public function __construct(array $tags, ?\DateTimeInterface $startDate, ?\DateTimeInterface $endDate)
+    {
+        $this->tags = $tags;
+        $this->startDate = $startDate;
+        $this->endDate = $endDate;
+    }
+
+    public static function createFromHttpRequest(Request $request): self
+    {
+        if (!$request->query->has('start-date') || !$request->query->has('end-date')) {
+            throw new \InvalidArgumentException('start-date or end-date arguments are missing');
+        }
+
+        $startDate = DateTimeFactory::create($request->get('start-date'), self::DATE_FORMAT);
+        $endDate = DateTimeFactory::create($request->get('end-date'), self::DATE_FORMAT);
+
+        $tags = $request->get('tags');
+
+        if (empty($tags)) {
```

Comment:
You can do directly `if ($tags = $request->get('tags'))`

Reply:
fixed!